### PR TITLE
Fix #7599: Adding/updating a property in settings is not used when property is a function

### LINKED ISSUE #7599: [8.0.0+] Adding/updating a property in settings is not used when property is a function
### Description
When updating a function in the settings object after initializing the grid, the new function is not used.

### Steps to reproduce
1. Initialize a grid using the :settings prop
2. set a new function on the settings object

### Demo

https://jsfiddle.net/541q03zp/
The issue appears to be located here : https://github.com/handsontable/vue-handsontable-official/blob/702231fea496e8824e23fb976fc4101ee7ddc542/src/helpers.ts#L247

The function simpleEqual cannot detect differences between functions.

### Your environment
* Vue wrapper version: 5.1.0
* Handsontable version: 8.3.1
* Browser Name and version: Chrome/88.0.4324.190
* Operating System: Windows 10

### PR DESCRIPTION
**Pull request by MiniGab**
### Context
When updating a function in the settings object after initializing the grid, the new function is not used.

This PR correct the simpleEqual method in helpers.js

This is a new PR for the closed : https://github.com/handsontable/vue-handsontable-official/pull/212

### How has this been tested?
The following demos highlight the issue and correction :

@handsontable/vue 4.1.1: https://jsfiddle.net/Lxdc950k/2/ 👍
@handonstable/vue 5.1.0 https://jsfiddle.net/541q03zp/ 👎
this version : https://jsfiddle.net/f219rvu6/ 👍
The PR also add a unit test.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. https://github.com/handsontable/handsontable/issues/7599

### Affected project(s):
- [ ] `handsontable`
- [ ] `@handsontable/angular`
- [ ] `@handsontable/react`
- [x] `@handsontable/vue`

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [x] I have signed the [Contributor License Agreement](https://docs.google.com/forms/d/e/1FAIpQLScpMq4swMelvw3-onxC8Jl29m0fVp5hpf7d1yQVklqVjGjWGA/viewform?c=0&w=1)
- [ ] My change requires a change to the documentation.

### CODE REVIEW

File: wrappers/vue/src/helpers.ts [hunk 1]
```diff
@@ -245,5 +245,9 @@ export function createVueComponent(vNode: VNode, parent: Vue, props: object, dat
  * @returns {boolean} `true` if they're the same, `false` otherwise.
  */
 function simpleEqual(objectA, objectB) {
+  if (typeof objectA === 'function' && typeof objectB === 'function') {
+    return ''+objectA === ''+objectB;
```
Comment (wszymanski) [inline wrappers/vue/src/helpers.ts:249]:
```suggestion
    return '' + objectA === '' + objectB;
```

**Review by wszymanski — CHANGES_REQUESTED** [review]
Hello @MiniGab. Thanks for your effort. It works, it's tested 🙂  Good job! I've just two small suggestions related to style of coding. Could you change it?

**Comment by MiniGab** [comment]
Hello @wszymanski. Thank you for the feedback. I did update the PR with your suggested changes :) !

**Comment by wszymanski** [comment]
@MiniGab Can I ask you for one more favor? Please add new `7599.json` file to `.changelogs/` directory (at the root of the repository). I'm pasting file content below:

```json
{
  "title": "Adding/updating a property in settings is not used when property is a function",
  "type": "fixed",
  "issue": 7599,
  "breaking": false,
  "framework": "vue"
}
```

**Comment by MiniGab** [comment]
@wszymanski Done !

**Review by wszymanski — APPROVED** [review]
Good job 👏 There is one more QA step, @aninde will check how code works. After her review, if everything is okey, the PR should be merged.

**Comment by aninde** [comment]
@MiniGab  thank you so much for putting your work into solving this problem and thus improving the quality of Handsontable-Vue.

My tests showed that this fix does not fully correct this whole regression bug because the dataset is not updated.
You can see it on the demos and gifs.

In v4.1.0 it didn't work quite well either.
We need to internally determine if we accept this partial fix for this bug and open a new regression bug on the updated data set itself. 

v4.1.1 https://jsfiddle.net/aninde/nsad42fp/
broken v5.1.0 https://jsfiddle.net/aninde/q8uLnsv0/
this fix https://jsfiddle.net/aninde/neth42yz/



5 1 0broken

this

**Review by aninde — COMMENTED** [review]
@MiniGab  thank you so much for putting your work into solving this problem and thus improving the quality of Handsontable-Vue.

My tests showed that this fix does not fully correct this whole regression bug because the dataset is not updated.
You can see it on the demos and gifs.

In v4.1.0 it didn't work quite well either.
We need to internally determine if we accept this as a partial fix for this bug.

v4.1.1 https://jsfiddle.net/aninde/nsad42fp/
broken v5.1.0 https://jsfiddle.net/aninde/q8uLnsv0/
this fix https://jsfiddle.net/aninde/neth42yz/



5 1 0broken

this

**Comment by AMBudnik** [comment]
The following pull request hasn't been merged as there are some missing improvements that @aninde previously mentioned.
I wouldn't want to ask you for more help (as you have already done a lot) @MiniGab. But I just want to know if you would be interested in making any changes to the following pull requests in the nearest future.

**Comment by AMBudnik** [comment]
I think that we can close this pull request as there are no updates since July. But feel free to leave a comment and I'll reopen the following pull request.

**Pull request closed without merging**